.github/workflows/tutorials.yml

```diff
@@ -0,0 +1,97 @@
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Prevent all build to stop if a single one fails
+      fail-fast: false
+      matrix:
+        name: [
+          ubuntu-18.04-gcc-9,
+          macOS-10.14-xcode-11.0
+          ]
+
+        include:
+
+          - name: ubuntu-18.04-gcc-9
+            os: ubuntu-18.04
+            compiler: gcc
+            version: "9"
+
+          - name: macOS-10.14-xcode-11.0
+            os: macOS-10.14
+            compiler: xcode
+            version: "11"
+
+        python-version: ['3.6', '3.7']
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install (Linux)
+      if: runner.os == 'Linux'
+      run: |
+        # Required for libc6-dbg:i386 and g++-multilib packages which are
+        # needed for x86 builds.
+        sudo dpkg --add-architecture i386
+        # clang-3.7 and earlier are not available in Bionic anymore so we get
+        # them from the Xenial repositories instead.
+        sudo add-apt-repository "deb http://dk.archive.ubuntu.com/ubuntu/ xenial main"
+        sudo add-apt-repository "deb http://dk.archive.ubuntu.com/ubuntu/ xenial universe"
+        # LLVM 9 is not in Bionic's repositories so we add the official LLVM repository.
+        if [ "${{ matrix.compiler }}" = "clang" ] && [ "${{ matrix.version }}" = "9" ]; then
+          sudo add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+        fi
+
+        sudo apt-get update
+
+        if [ "${{ matrix.compiler }}" = "gcc" ]; then
+          sudo apt-get install -y g++-${{ matrix.version }} g++-${{ matrix.version }}-multilib
+          echo "::set-env name=CC::gcc-${{ matrix.version }}"
+          echo "::set-env name=CXX::g++-${{ matrix.version }}"
+        else
+          sudo apt-get install -y clang-${{ matrix.version }} g++-multilib
+          echo "::set-env name=CC::clang-${{ matrix.version }}"
+          echo "::set-env name=CXX::clang++-${{ matrix.version }}"
+        fi
+
+        sudo apt-get install -y mpich libmpich-dev
+
+    - name: Install (macOS)
+      if: runner.os == 'macOS'
+      run: |
+        brew install mpich
+        if [ "${{ matrix.compiler }}" = "gcc" ]; then
+          brew install gcc@${{ matrix.version }}
+          echo "::set-env name=CC::gcc-${{ matrix.version }}"
+          echo "::set-env name=CXX::g++-${{ matrix.version }}"
+        else
+          sudo xcode-select -switch /Applications/Xcode_${{ matrix.version }}.app
+          echo "::set-env name=CC::clang"
+          echo "::set-env name=CXX::clang++"
+        fi
+
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip
+        pip install -e .[extras]
+
+    - name: Evaluate cells of example notebooks
+      run: |
+        bash .github/workflows/test-notebooks.sh $CC
```

Comment:
I don't particularly like that, I would rather keep the  purest commands here for clarity. Can even make multiple steps 
```
- name: Run cfd notebook
  run: py.test .... cfd

- name : Run seismic ....
...
...

```

That would be nicer.

This is would also make it a lot clearer just looking at the log what passed/failed rather than a single bash script for everything.

Reply:
I agree, this would make each test more atomic, but it was discussed with @navjotk that would be nice to decouple into a separate file (`test-notebooks.sh`) such that it can be used potentially in conjunction with #1071

Reply:
I disagree, what docker wants to do should not impact this action. It's a standalone action and the sybthax above makes it clearer and easier to debug and maintain. If 1071 needs scripts it can make some when needed and not sure it's the best way neither